Remote: don't print tracebacks for Error exceptions handled downstream

enkore opened the pull request:
(GH wanted to put this into master before)

ThomasWaldmann commented on borg/remote.py:
```diff
@@ -93,8 +94,11 @@ def serve(self):
-                        logging.error(sysinfo())
+                        if not isinstance(e, (Repository.DoesNotExist, Repository.AlreadyExists, Repository.CheckNeeded,
+                                              IntegrityError, PathNotAllowed, Repository.ObjectNotFound,
+                                              InvalidRPCMethod)):
```
are these the ones subclassing Error (not ErrorWithTraceback)?

ThomasWaldmann replied:
and if so, could we just check for that?

enkore replied:
No, these are the special-cased exceptions handled [here](https://github.com/enkore/borg/blob/issue/792/borg/remote.py#L234). These are the only Errors that the error-handling code in archiver.py:main will see.

enkore replied:
I'll toss a note in about the relation between these two code blocks

enkore commented:
I updated this after some more testing and removed some of the exceptions handled by the client, namely all deriving from ErrorWithTraceback, who should (also) print a (remote) traceback.
